x/docs/internal/keeper/querier.go

```diff
@@ -58,3 +62,40 @@ func queryGetSentDocuments(ctx sdk.Context, path []string, keeper Keeper) (res [
 
 	return bz, nil
 }
+
+func queryGetReceivedDocsReceipts(ctx sdk.Context, path []string, keeper Keeper) (res []byte, err sdk.Error) {
+	addr := path[0]
+	address, _ := sdk.AccAddressFromBech32(addr)
+
+	receipts := keeper.GetUserReceivedReceipts(ctx, address)
+	if receipts == nil {
+		receipts = make([]types.DocumentReceipt, 0)
+	}
+
+	bz, err2 := codec.MarshalJSONIndent(keeper.cdc, receipts)
+	if err2 != nil {
+		return nil, sdk.ErrUnknownRequest("Could not marshal result to JSON")
+	}
+
+	return bz, nil
+
+}
+
+func queryGetReceiptByUuid(ctx sdk.Context, path []string, keeper Keeper) (res []byte, err sdk.Error) {
```

Comment:
I think this method and the above one could be merged into a single what that checks the length of the `path` params and acts accordingly to the case where it has only 1 element inside or when it has 2.